src/ConfigCatProvider.tsx

```diff
@@ -15,39 +17,33 @@ type ConfigCatProviderProps = {
 };
 
 type ConfigCatProviderState = {
-  client: IConfigCatClient;
+  client?: IConfigCatClient;
```

Comment:
I think it would be preferable to avoid making `client` nullable if possible. Apart from it being an annoying breaking change, also degrades the ergonomics of the SDK's public API.

Instead, how about checking if the code is running in a browser environment (e.g. by [checking for the availability of the `window`](https://stackoverflow.com/a/34550964) or `XMLHttpRequest` object) and when it's not, then using some fake implementation of `IConfigCatClient` (something like [this](https://github.com/codedbychavez/configcat-vue/blob/07fa94521a03cb6e68b9221dd6b51bc6005e8bd9/tests/mocks/ConfigCatClientMocks.ts#L8-L82))?

This way probably we wouldn't even need to move the initialization logic to `componentDidMount`.